lib/queue.js

```diff
@@ -16,6 +16,29 @@ var allowedStatistics = [
     "actualDataSize"
 ];
 
+function noop() {}
+
+/**
+ * Recursive function that compares immutable properties from the first object
+ * to the same properties in the second one and returns true if all of the
+ * properties matched. If not, it returns false.
+ */
+function compare (a, b) {
+    var matches;
+
+    for (var key in a) {
+        if (a.hasOwnProperty(key)) {
+            if (typeof a[key] === "object" && typeof b[key] === "object") {
+                matches = compare(a[key], b[key]);
+            } else {
+                matches = matches === false ? matches : a[key] === b[key];
```

Comment:
Yeah, you're probably right! I think the thought might've struck me, but I skipped it since I figured the comparator object should always be fairly small. But it'd probably smarter to just optimize for performance, especially for a project like this that serves as the base for other ones.